### PR TITLE
MDOT: Analytics

### PR DESCRIPTION
**Pull request by maharielrosario**
## Description
This PR focuses on adding analytics to areas of interest in the form.

## Screenshots
Screen Shot 2020-05-19 at 1 38 49 PM

## Acceptance criteria
- [ ]  Analytics should be sent on submission, successful submissions, failed submissions, and ineligibility orders

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/2346/config/form.js [hunk 1]
```diff
@@ -45,12 +45,59 @@ const formPageTitlesLookup = {
 
 const addressSchema = buildAddressSchema(true);
 
+const asyncReturn = (returnValue, error, delay = 300) =>
+  new Promise((resolve, reject) => {
+    setTimeout(() => {
+      const randomNumber = Math.round(Math.random() * 10);
```
Comment (mr0sari0) [inline src/applications/disability-benefits/2346/config/form.js:51]: Adding a temporary way to simulate whether the call was successful or not.

**Comment by bradl3yC** [comment]
Instead of pushing directly to the dataLayer, we need to leverage 
```
import recordEvent from 'platform/monitoring/record-event'
```

File: src/applications/disability-benefits/2346/config/form.js [hunk 2]
```diff
@@ -45,12 +45,59 @@ const formPageTitlesLookup = {
...
+  const onSuccess = resp =>
+    new Promise(resolve => {
+      window.dataLayer.push({
+        event: 'bam-2346a-submission-successful',
+        'bam-quantityOrdered': itemQuantities,
+      });
+      return resolve(resp);
+    });
+
+  const onFailure = error =>
```
Comment (bradl3yC) [inline src/applications/disability-benefits/2346/config/form.js:77]: I would prefer to see a try catch used as opposed to .then/.catch

**Review by bradl3yC — CHANGES_REQUESTED** [review]
This is g2g as soon as you use the platform helper - swapping to async await in a try/catch isn't necessary as we're probably removing that code at some point

**Comment by mr0sari0** [comment]
> This is g2g as soon as you use the platform helper - swapping to async await in a try/catch isn't necessary as we're probably removing that code at some point

Of course I see this after I finish refactoring.  Anyway, I'm running into a bug where my request is getting its own value apart from what I'm returning.

**Comment by mr0sari0** [comment]
> Instead of pushing directly to the dataLayer, we need to leverage
> 
> ```
> import recordEvent from 'platform/monitoring/record-event'
> ```

Just updated.

**Comment by bradl3yC** [comment]
Can you attach screens of the data later with the tags working?

**Review by bradl3yC — APPROVED** [review]

**Pull request merged**